### PR TITLE
Docs for setting up haskell-language-server & your favorite editor

### PR DESCRIPTION
**Pull request by 0xtmphey**
Hey folks!

Here are some instructions how you can setup IDE-like features (auto-completion, type hints, go-to-definition) for Haskell/Plutus development.

I've added instructions for Linux & macOS for following editors
- VSCode
- Sublime Text
- Emacs
Vim is missing here since i've never set up vim with LSP.

Also we're missing instructions for Windows.

Feedback appreciated!

### CODE REVIEW

File: docs/editors/emacs.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+   (require 'lsp)
+   (require 'lsp-haskell)
+   ;; Hooks so haskell and literate haskell major modes trigger LSP setup
+   (add-hook 'haskell-mode-hook #'lsp)
+   (add-hook 'haskell-literate-mode-hook #'lsp)
+   ```
+6. Close the editor and re-open it in the project folder
+   ```bash
+   cd <your-projects-dir>/plutus-pioneer-program/code/week02
+   emacs .
```
Comment (Dauta) [inline docs/editors/emacs.md]: I think it's not really necessary to specify this path. Something like 
```
cd <your-projects-dir> emacs .
```
should be enough, I think :)

Also applies to most other entries as well.

**Review by Dauta — COMMENTED** [review]
Hi @tmphey, thanks for your contribution! Looks great, just a small suggestion.

File: docs/editors/emacs.md (hunk 1 above)
Comment (0xtmphey) [reply docs/editors/emacs.md]: Thanks @Dauta , I'll fix it!

Comment (0xtmphey) [reply docs/editors/emacs.md]: Fixed @Dauta

**Review by Dauta — APPROVED** [review]
Awesome! 👍

**Pull request merged**